qemu/tests/migration_with_vsock.py

```diff
@@ -0,0 +1,118 @@
+import random
+import logging
+import ast
+
+from avocado.utils import process
+from virttest import error_context
+from virttest import utils_misc
+from qemu.tests.vsock_test import (
+    compile_nc_vsock,
+    nc_vsock_listen,
+    send_data_from_guest_to_host,
+    check_received_data,
+    nc_vsock_connect,
+    check_guest_nc_vsock_exit,
+)
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Vsock migration test
+
+    1. Boot guest with vhost-vsock-pci device
+    2. Download and compile nc-vsock on both guest and host
+    S1):
+    3. Start listening inside guest, nc-vsock -l $port
+    4. Connect guest CID from host, nc-vsock $guest_cid $port
+    5. Input character, e.g. 'Hello world'
+    6. Check if guest receive the content correctly
+    7. Reboot guest with differnt guest CID
+    8. Migration
```

Comment:
From the test case,  both host and guest vsock socket will disconnected after migrate start.  This test scenario is missed. 
@michiboo @qcheng-redhat Please check this.